src/zfile.c

```diff
@@ -101,9 +107,42 @@ zfile_new (const char *path, const char *name)
     }
     self->handle = 0;
     zfile_restat (self);
+    self->fd = -1;
+    self->close_fd = false;
     return self;
 }
 
+//  --------------------------------------------------------------------------
+//  Constructor
+//  Create new temporary file for writing via tmpfile. File is automaticaly
+//  deleted on destroy
+
+zfile_t *
+zfile_tmp (void)
+{
+    zfile_t *self = (zfile_t *) zmalloc (sizeof (zfile_t));
+    assert (self);
+    self->remove_on_destroy = true;
+
+    char buffer [PATH_MAX];
+    strcpy (buffer, "/tmp/czmq_zfile.XXXXXX");
+    self->fd = mkstemp (buffer);
```

Comment:
Sorry, this time it's Windows turn to be unhappy :-(

```
C:\projects\czmq\src\zfile.c(127): error C2065: 'PATH_MAX': undeclared identifier [C:\projects\build_czmq\czmq.vcxproj]
C:\projects\czmq\src\zfile.c(129): error C3861: 'mkstemp': identifier not found [C:\projects\build_czmq\czmq.vcxproj]
```

Reply:
I know - I tried to get a green even on Windows. Is there any way I can test Windows build without making PR? I am afraid that I don't have Windows machine nearby ...